elevation_mapping_costmap_2d_plugin/include/elevation_mapping_costmap_2d_plugin/elevation_layer.hpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * elevation_layer.h
+ *
+ *  Created on: Nov 5, 2018
+ *      Author: Eugenio Chisari
+ *	 Institute: ANYbotics
+ */
+
+#pragma once
+
+// C++ Libraries
+#include <atomic>
+#include <boost/algorithm/string/predicate.hpp>
+#include <mutex>
+
+// Ros headers
+#include <dynamic_reconfigure/server.h>
+#include <nav_msgs/OccupancyGrid.h>
+#include <ros/ros.h>
+
+// Costmap2d headers
+#include <costmap_2d/costmap_layer.h>
+#include <costmap_2d/footprint.h>
+#include <costmap_2d/layered_costmap.h>
+#include <costmap_2d/observation_buffer.h>
+
+// Package related headers
+#include <filters/filter_chain.h>
+#include <message_filters/subscriber.h>
+#include <grid_map_ros/GridMapRosConverter.hpp>
+#include "elevation_mapping_costmap_2d_plugin/ElevationPluginConfig.h"
+
+namespace elevation_mapping_costmap_2d_plugin {
+
+/*!
+ * Method to update the cost of a portion of map
+ */
+enum CombinationMethod { Overwrite, Maximum, Unknown };
+
+/*!
+ * converts the string from the yaml file to the proper CombinationMethod enum
```

Comment:
All comments should start with a capital letter and end with a period, see coding guidelines.